### DISCUSSION TITLE
Using a variable as the GitHub action version

### DISCUSSION THREAD

**Discussion by JJ** [community/community#108543] [Actions] [answered]
### Select Topic Area

Question

### Body

Suppose you want to check in a PR or a push the GitHub action in the version that's in the actual branch. You can use

```
uses: mygithub/action@my-branch-name
```

But you would have to change the branch name every time you do that; if you want to test github actions, that gets tired after a while.

You can't use `@HEAD` because that would be the head of the repository. Github keeps the ref to the branch itself as a variable, so it would be ideal to do something like:

```yaml
mygithub/action@${{github.head_ref || github.ref_name }}
```

That, however, fails with

```
unrecognized named-value: 'github'. Located at position 1 within expression: github.head_ref || github.ref_name
```

So, long story short, is there an unambiguous way to version the github action to always refer to the branch it's in, or otherwise make a github action use the repo itself, in the current branch?

**Comment by spenserblack** [comment]
Have you tried `uses: ./`?

**Reply by JJ** [reply]
Would that need checkout first?

**Reply by spenserblack** [reply, marked as answer]
Oh, yes, that's right, I believe you would need to use `actions/checkout` first. Additionally, if your action needs to be built (like a TypeScript action), you would have to make sure you've committed the latest build to match your source code.

**Comment by brucellino** [comment]
I cam here to ask this. In my case, I can't use the repo itself. I need to refer to an existing action:

```
jobs:
  build:
    strategy:
      matrix:
        version:
          - v5.22.2
          - v24.2.1
    name: Ansible Lint ${{ matrix.version }}
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v4
      - name: Run ansible-lint
        uses: ansible/ansible-lint@v24.2.1
```

**Comment by jpmaniqis** [comment]
> ### Select Topic Area
> 
> Question
> 
> ### Body
> 
> Suppose you want to check in a PR or a push the GitHub action in the version that's in the actual branch. You can use
> 
> 
> 
> ```
> 
> uses: mygithub/action@my-branch-name
> 
> ```
> 
> 
> 
> But you would have to change the branch name every time you do that; if you want to test github actions, that gets tired after a while.
> 
> 
> 
> You can't use `@HEAD` because that would be the head of the repository. Github keeps the ref to the branch itself as a variable, so it would be ideal to do something like:
> 
> 
> 
> ```yaml
> 
> mygithub/action@${{github.head_ref || github.ref_name }}
> 
> ```
> 
> 
> 
> That, however, fails with
> 
> 
> 
> ```
> 
> unrecognized named-value: 'github'. Located at position 1 within expression: github.head_ref || github.ref_name
> 
> ```
> 
> 
> 
> So, long story short, is there an unambiguous way to version the github action to always refer to the branch it's in, or otherwise make a github action use the repo itself, in the current branch?
